core/src/main/java/org/elasticsearch/action/admin/indices/analyze/AnalyzeRequest.java

```diff
@@ -86,6 +91,7 @@ public String tokenizer() {
     }
 
     public AnalyzeRequest tokenFilters(String... tokenFilters) {
+        assert tokenFilters != null: "token filters must not be null";
```

Comment:
IMO I think this should be an if statement that throws an `IllegalArguementException` instead of an assert since a user can easily pass in null at runtime here. That way they will get an exception when its set rather than a weird exception happening when we try to use the variable. Same goes for `charFilters` and `attributes` below